src/cli.rs

```diff
@@ -151,16 +158,35 @@ fn resynth() -> Result<(), ()> {
     };
     let mut stdout = StandardStream::stdout(color);
 
-    let mut out = argv.value_of("out").map_or(
-        PathBuf::new(),
-        PathBuf::from,
-    );
+    let use_filenames = argv.contains_id("out");
 
-    for arg in argv.values_of("in").unwrap() {
-        let p = Path::new(arg);
+    let in_args = argv.get_many::<String>("in").unwrap();
 
-        out.push(p.file_stem().unwrap());
-        out.set_extension("pcap");
+    let out_args = argv.get_many::<String>("out").unwrap_or_default().collect::<Vec<_>>();
+    let out_dir = argv.get_one::<String>("outdir");
+
+    if use_filenames && out_args.len() != in_args.len() {
+        cmd.error(
+            ErrorKind::WrongNumberOfValues,
+            format!(
+                "Received {} output(s), expected: {}",
+                out_args.len(),
+                in_args.len(),
+            ),
+        )
+        .exit();
+    }
+
+    for (i, input) in in_args.enumerate() {
+        let p = Path::new(input);
+        let mut out: PathBuf;
```

Comment:
You can avoid `let mut` bindings in these cases by using if-expressions, eg.

```rust
let out = if whatever {
    PathBuf::from(out_args[i])
} else {
    let p = PathBuf::from(out_dir.unwrap());
    out.push(p.file_stem().unwrap());
    out.set_extension("pcap");
    out
};
```

You also get type-inference then, too, which helps avoid boilerplate.

Reply:
Much tidier, thank you!